arduino-ide-extension/src/browser/contributions/survey-notification.ts

```diff
@@ -0,0 +1,62 @@
+import { MessageService } from '@theia/core';
+import { FrontendApplicationContribution } from '@theia/core/lib/browser';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { shell } from '@theia/electron/shared/electron';
+import { LocalStorageService } from '@theia/core/lib/browser';
+import { SurveyRetriever } from '../survey/survey-retriever';
+import { nls } from '@theia/core/lib/common';
+
+const SURVEY_MESSAGE = nls.localize(
+  'arduino/survey/surveyMessage',
+  'Please help us improve by answering this super short survey. We value our community and would like to get to know our supporters a little better.'
+);
+const DO_NOT_SHOW_AGAIN = nls.localize(
+  'arduino/survey/dismissSurvey',
+  'DON’T SHOW ANYMORE'
+);
+const GO_TO_SURVEY = nls.localize(
+  'arduino/survey/answerSurvey',
+  'ANSWER SURVEY'
+);
+
+@injectable()
+export class SurveyNotification implements FrontendApplicationContribution {
+  @inject(MessageService)
+  protected readonly messageService: MessageService;
+
+  @inject(LocalStorageService)
+  protected readonly localStorageService: LocalStorageService;
+
+  @inject(SurveyRetriever)
+  protected readonly surveyRetriever: SurveyRetriever;
+
+  async onStart(): Promise<void> {
+    const survey = await this.surveyRetriever.getSurvey();
+    const surveyAnswered = await this.localStorageService.getData(
+      this.surveyKey(survey.id),
+      'notAnswered'
+    );
+
+    if (surveyAnswered !== 'notAnswered') {
```

Comment:
Why not check against `undefined`?

As I see, the current possible values are `true |  false | 'notAnswered'`, why not `true | false  | undefined`?